### PR TITLE
12.0 [MIG] maintenance_plan_activity

### PR DESCRIPTION
**Pull request by NuriaXifre**
Migration of the module maintenance_plan_activity to v12

### CODE REVIEW

**Comment by etobella** [comment]
Well, I think we should need to migrate the improvements of plan #19 and then fix this module, also, we should need to fix the addon on 11.0, as functions have changed.

**Comment by etobella** [comment]
Well, now it is time to make it compatible with #60 :smile: and integrate with #61

**Comment by etobella** [comment]
All PRs have been migrated. Can you rebase and cherry pick in order to test them?

**Review by etobella — APPROVED** [review]

File: maintenance_plan_activity/tests/test_maintenance_plan_activity.py [hunk 1]
```diff
@@ -49,11 +49,11 @@ def test_01_cron_auto_create_activities(self):
         # Check if activity Call has been created for the request 1
         generated_activities = self.mail_activity_obj.search(
             [('res_id', '=', request_1.id)])
-        self.assertEqual(len(generated_activities), 1)
+        self.assertEqual(len(generated_activities), 2)
```
Comment (etobella) [inline maintenance_plan_activity/tests/test_maintenance_plan_activity.py:52]: Why there is two activities?

File: maintenance_plan_activity/tests/test_maintenance_plan_activity.py [hunk 2]
```diff
@@ -49,11 +49,11 @@ def test_01_cron_auto_create_activities(self):
...
         self.assertEqual(
             generated_activities[0].activity_type_id.name, self.call.name)
         self.assertEqual(
-            generated_activities[0].date_deadline,
+            str(generated_activities[0].date_deadline),
```
Comment (etobella) [inline maintenance_plan_activity/tests/test_maintenance_plan_activity.py]: It is simpler to keep this as it was and delete the following `fields.Date.to_string`

**Review by etobella — CHANGES_REQUESTED** [review]

File: maintenance_plan_activity/tests/test_maintenance_plan_activity.py (hunk 1 above)
Comment (NuriaXifre) [reply maintenance_plan_activity/tests/test_maintenance_plan_activity.py:52]: In v12, an activity " Maintenance Request" is created, that is why there are 2 activities.
Selection_090

File: maintenance_plan_activity/tests/test_maintenance_plan_activity.py (hunk 2 above)
Comment (NuriaXifre) [reply maintenance_plan_activity/tests/test_maintenance_plan_activity.py]: okey

File: maintenance_plan_activity/tests/test_maintenance_plan_activity.py (hunk 1 above)
Comment (etobella) [reply maintenance_plan_activity/tests/test_maintenance_plan_activity.py:52]: Thanks!! I didn't know :wink:

**Review by jarroyomorales — APPROVED** [review]
LGTM :+1:

**Review by etobella — APPROVED** [review]

**Comment by etobella** [comment]
I think now it is time for
/ocabot merge
Thanks @NuriaMForgeFlow

**Pull request merged**